public/app/features/explore/state/actions.ts

```diff
@@ -331,8 +350,7 @@ export function loadDatasource(exploreId: ExploreId, instance: DataSourceApi): T
       return;
     }
 
-    dispatch(loadDatasourceSuccess(exploreId, instance));
-    return Promise.resolve();
```

Comment:
Do we need the Promise.resolve()?

Reply:
no should not be needed.

Reply:
When I added Promise.reject then you get typescript errors if you don't also return Promise.resolve().
```typescript
public/app/features/explore/state/actions.ts:296:10 - error TS7030: Not all code paths return a value.

296   return async dispatch => {
             ~~~~~~~~~~~~~~~~~~~
```

Reply:
Changed to check if state has `datasourceError` before running any queries